### ISSUE TITLE
ImportError: cannot import name 'OpenAI' from 'openai'

### ISSUE THREAD

**Issue by nasosev** [PrefectHQ/marvin#701]
### First check

- [X] I added a descriptive title to this issue.
- [X] I used the GitHub search to try to find a similar issue and didn't find one.
- [X] I searched the Marvin documentation for this issue.

### Bug summary

Cannot run the first example in https://www.askmarvin.ai/welcome/quickstart/ because of an error.

### Reproduction

```python3
1. create a fresh python 3.11.5 environment with `python3.11 -m venv myenv`
2. activate env on macos `source myenv/bin/activate`
3. install marvin `pip install marvin`
4. in a python shell, run 'from openai import OpenAI'

Note the openai version is:


>>> import openai
>>> print(openai.VERSION)
0.28.1
```
```


### Error

```python3
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ImportError: cannot import name 'OpenAI' from 'openai' (/Users/naso/Documents/GitHub/test/myenv/lib/python3.11/site-packages/openai/__init__.py)
>>>
```

### Versions

```Text
Version:		1.5.6
Python version:		3.11.5
OS/Arch:		darwin/x86_64
```

**Comment by marwan116** [comment]
This bug only shows as of marvin==1.5.6, due to this code change https://github.com/PrefectHQ/marvin/pull/634/files

It pins the openai dependency to a version less than v1.0.0

**Comment by marwan116** [comment]
Ah but seems like if openai is not pinned to <v.1.0

then you get an error when trying a marvin import

```
from marvin import ai_model
```

```
ModuleNotFoundError: No module named 'openai.openai_object' 
```

So it now makes sense why the code change was made for v1.5.6

Here is a working code example to unblock your discovery of the tool

```python
from marvin import ai_model
from pydantic import BaseModel, Field
import openai
import os

openai.api_key = os.environ.get("OPENAI_API_KEY")

@ai_model(client = openai)
class Location(BaseModel):
    city: str
    state_abbreviation: str = Field(
        ..., 
        description="The two-letter state abbreviation"
    )


Location("The Big Apple")
```

**Comment by nasosev** [comment]
> ```python
> from marvin import ai_model
> from pydantic import BaseModel, Field
> import openai
> import os
> 
> openai.api_key = os.environ.get("OPENAI_API_KEY")
> 
> @ai_model(client = openai)
> class Location(BaseModel):
>     city: str
>     state_abbreviation: str = Field(
>         ..., 
>         description="The two-letter state abbreviation"
>     )
> 
> 
> Location("The Big Apple")
> ```

Thank you very much. I've tried this, replacing the OPEN_AI_KEY with a key `sk-[REDACTED]` I've generated at https://platform.openai.com/api-keys, I get the following error below. Do you know what could be wrong? 

Thank you

```
Last login: Sat Dec 23 19:57:41 on ttys001
(base) naso:~/Documents/GitHub/test> python app.py
Traceback (most recent call last):
  File "/Users/naso/anaconda3/lib/python3.11/site-packages/marvin/utilities/async_utils.py", line 61, in run_sync
    loop = asyncio.get_running_loop()
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
RuntimeError: no running event loop

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/naso/Documents/GitHub/test/app.py", line 17, in <module>
    Location("The Big Apple")
  File "/Users/naso/anaconda3/lib/python3.11/site-packages/marvin/components/ai_model.py", line 75, in __init__
    kwargs.update(self.__class__.call(text, instructions=instructions_))
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/naso/anaconda3/lib/python3.11/site-packages/marvin/components/ai_model.py", line 281, in call
    .create()
     ^^^^^^^^
  File "/Users/naso/anaconda3/lib/python3.11/site-packages/marvin/core/ChatCompletion/abstract.py", line 152, in create
    response_data = self._send_request(**serialized_request)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/naso/anaconda3/lib/python3.11/site-packages/marvin/core/ChatCompletion/providers/openai.py", line 198, in _send_request
    return run_sync(
           ^^^^^^^^^
  File "/Users/naso/anaconda3/lib/python3.11/site-packages/marvin/utilities/async_utils.py", line 69, in run_sync
    return asyncio.run(coroutine)
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/naso/anaconda3/lib/python3.11/asyncio/runners.py", line 190, in run
    return runner.run(main)
           ^^^^^^^^^^^^^^^^
  File "/Users/naso/anaconda3/lib/python3.11/asyncio/runners.py", line 118, in run
    return self._loop.run_until_complete(task)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/naso/anaconda3/lib/python3.11/asyncio/base_events.py", line 653, in run_until_complete
    return future.result()
           ^^^^^^^^^^^^^^^
  File "/Users/naso/anaconda3/lib/python3.11/site-packages/marvin/core/ChatCompletion/providers/openai.py", line 211, in _send_request_async
    response = await openai.ChatCompletion.acreate(**serialized_request)  # type: ignore # noqa
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/naso/anaconda3/lib/python3.11/site-packages/openai/api_resources/chat_completion.py", line 45, in acreate
    return await super().acreate(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/naso/anaconda3/lib/python3.11/site-packages/openai/api_resources/abstract/engine_api_resource.py", line 216, in acreate
    ) = cls.__prepare_create_request(
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/naso/anaconda3/lib/python3.11/site-packages/openai/api_resources/abstract/engine_api_resource.py", line 108, in __prepare_create_request
    requestor = api_requestor.APIRequestor(
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/naso/anaconda3/lib/python3.11/site-packages/openai/api_requestor.py", line 139, in __init__
    self.api_key = key or util.default_api_key()
                          ^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/naso/anaconda3/lib/python3.11/site-packages/openai/util.py", line 186, in default_api_key
    raise openai.error.AuthenticationError(
openai.error.AuthenticationError: No API key provided. You can set your API key in code using 'openai.api_key = <API-KEY>', or you can set the environment variable OPENAI_API_KEY=<API-KEY>). If your API key is stored in a file, you can point the openai module at it with 'openai.api_key_path = <PATH>'. You can generate API keys in the OpenAI web interface. See https://platform.openai.com/account/api-keys for details.
```

**Comment by daviddwlee84** [comment]
Same issue here.

If `pip install openai --upgrade` then it will cause another error

`ModuleNotFoundError: No module named 'openai.openai_object'` (this seems removed in openai v1)

```txt
Traceback (most recent call last):
  File "C:\Users\david\AppData\Roaming\Python\Python311\site-packages\streamlit\runtime\scriptrunner\script_runner.py", line 534, in _run_script
    exec(code, module.__dict__)
  File "C:\Users\david\Documents\Program\MarvinPlayground\pages\0_AI_Function.py", line 11, in <module>
    from marvin import ai_fn
  File "c:\python311\Lib\site-packages\marvin\__init__.py", line 3, in <module>
    from .components import (
  File "c:\python311\Lib\site-packages\marvin\components\__init__.py", line 3, in <module>
    from .ai_application import AIApplication
  File "c:\python311\Lib\site-packages\marvin\components\ai_application.py", line 9, in <module>
    from marvin.core.ChatCompletion.providers.openai import get_context_size
  File "c:\python311\Lib\site-packages\marvin\core\ChatCompletion\providers\openai.py", line 10, in <module>
    from openai.openai_object import OpenAIObject
ModuleNotFoundError: No module named 'openai.openai_object'
```

**Comment by daviddwlee84** [comment]
`pip install marvin==2.0.0a1`

> support for OpenAI 1.X Client and migration to Pydantic V2.

[Release v2.0.0 pre-release · PrefectHQ/marvin](https://github.com/PrefectHQ/marvin/releases/tag/v2.0.0-alpha.2)

I think the documentation (e.g. [Quickstart - Marvin](https://www.askmarvin.ai/welcome/quickstart/#__tabbed_1_1)) is already using v2.0 syntax.

**Comment by anjor** [comment]
is there an openai version that works with 1.5.6 or should 1.5.6 be patched? I tried with the default (0.28.1) as well as 1.6.1, and like mentioned above 0.28.1 fails to find `OpenAI` and 1.6.1 fails to find `openai_object`

**Comment by zzstoatzz** [comment]
hi @daviddwlee84 and @anjor - apologies about the drift in releases / docs.

you can find docs on how to use marvin with azure [here](https://www.askmarvin.ai/docs/configuration/settings/#settings-for-using-azure-openai-models)

let us know if you run into any issues
